test/chains/icon/localnet.go

```diff
@@ -436,53 +439,59 @@ func (c *IconLocalnet) GetIBCAddress(key string) string {
 }
 
 func (c *IconLocalnet) ConfigureBaseConnection(ctx context.Context, connection chains.XCallConnection) (context.Context, error) {
-	temp := "07-tendermint-0"
-	params := `{"connectionId":"` + connection.ConnectionId + `","counterpartyPortId":"` + connection.CounterPartyPortId + `","counterpartyNid":"` + connection.CounterpartyNid + `","clientId":"` + temp + `","timeoutHeight":"1000"}`
-	ctx, err := c.executeContract(ctx, c.IBCAddresses["connection"], connection.KeyName, "configureConnection", params)
+	testcase := ctx.Value("testcase").(string)
+	clientId := "07-tendermint-0"
+	params, _ := json.Marshal(map[string]string{
+		"connectionId":       connection.ConnectionId,
+		"counterpartyPortId": connection.CounterPartyPortId,
+		"counterpartyNid":    connection.CounterpartyNid,
+		"clientId":           clientId,
+		"timeoutHeight":      strconv.Itoa(connection.TimeoutHeight),
+	})
+	connectionAddress := c.IBCAddresses[fmt.Sprintf("connection-%s", testcase)]
+	//params := `{"connectionId":"` + connection.ConnectionId + `","counterpartyPortId":"` + connection.CounterPartyPortId + `","counterpartyNid":"` + connection.CounterpartyNid + `","clientId":"` + temp + `","timeoutHeight":"100"}`
+	ctx, err := c.executeContract(context.Background(), connectionAddress, connection.KeyName, "configureConnection", string(params))
 	if err != nil {
-		return nil, err
+		panic(err)
```

Comment:
What is the rule you used for panic vs return?

Reply:
Instead of panic, it should have returned error itself. I'll fix it and commit new changes.